[UI] Send: Manual Control 2

ichthus1604 opened the pull request:
WIP.

New PR to minimize changes made in previous one.

![image](https://github.com/zkSNACKs/WalletWasabi/assets/98904713/a21da92e-3224-407c-b34f-1abf4300dabc)

 - Includes some minimal cleanup to various ViewModels in the Send flow (nothing too disruptive)
 - Implements the Manual Control screen as seen in the screenshot
 - the Max amount in the send screen is affected by this (only max amount as per selected coins, or wallet balance for regular flow)
 - the Privacy Suggestions are affected (only operate with the selected coins and not full wallet coins)
 - This PR depends on #12841. I have added a temporary additional "Manual Control" as a regular button beside "Send" in the main wallet UI. This is temporary until the `SubActionButton` is available.
 - Marked as draft until the above PR is merged.
 - @soosr in the meantime you might want to give an initial review for this PR.

soosr commented:
Don't forget to add the select all button.
![image](https://github.com/zkSNACKs/WalletWasabi/assets/16364053/946d3066-d730-4eaa-892d-8bc7915912f9)

soosr commented:
@ichthus1604 
https://github.com/zkSNACKs/WalletWasabi/pull/12841 is merged now, you can use the button.

ichthus1604 commented:
@soosr 

> The amount isn't centralized and its look should match the Figma design.

Fixed.

> Navigating back should be possible from Send.

Fixed.

> After restringing the coins in the manual control screen, on the Transaction Preview when opening the Review Coins dialog, only the restricted coins should be in the list.

Fixed.

> #12841 is merged now, you can use the button.

Done.

soosr commented:
- It seems like the label management still uses all the wallet coins even if it was restricted.
![image](https://github.com/zkSNACKs/WalletWasabi/assets/16364053/21660f59-2c61-4ce4-bdd0-ac5225529878)

- Position of the dialog buttons changed.

ichthus1604 commented:
@soosr 

> Label management still uses all the wallet coins even if it was restricted

Implemented this. I'm not sure it's working 100% correctly, because I don't have a wallet with many different labels to test with. Can you confirm?

soosr commented:
> @soosr
> 
> > Label management still uses all the wallet coins even if it was restricted
> 
> Implemented this. I'm not sure it's working 100% correctly, because I don't have a wallet with many different labels to test with. Can you confirm?

Works fine!

soosr reviewed (changes requested):
- The position of Back, Cancel, and Continue buttons on the dialog is wrong.
- The Amount background is visible when there is no selection.
![image](https://github.com/zkSNACKs/WalletWasabi/assets/16364053/30dffb83-15d8-4ade-84d4-192a10afe89b)
- When the Amount become visible, the position of Cancel and Continue button shifts.
- The default option should be also visible in the SubActions.
   - See Figma for designs and icons. 
   - https://www.figma.com/file/NykhItqdw5zQ4t4K9yBnkn/2.0.8?type=design&node-id=0-1&mode=design&t=wqpqwpSzN51Yl4lP-0

soosr commented:
@ichthus1604 Is there any blocker for keeping the original positions (as on master) of the cancel, continue button on the dialogs?

ichthus1604 commented:
@soosr 

> Is there any blocker for keeping the original positions (as on master) of the cancel, continue button on the dialogs?

Fixed. I had fixed vertical margins, but I didn't realize horizontal margins were also wrong.

soosr commented:
@ichthus1604 Please fix conflicts

soosr reviewed (changes requested):
Tested and works as expected.

I will review the code after the remaining minor things are fixed:
- Select all quick action is missing.
![image](https://github.com/WalletWasabi/WalletWasabi/assets/16364053/48864aa0-cea3-4375-b33e-ffef8d01157a)

- The Automatic icon is different (you might have seen an old version of designs)
   - https://www.figma.com/design/NykhItqdw5zQ4t4K9yBnkn/2.0.8?node-id=0-1&t=pURqReD17shr6BZD-0
![image](https://github.com/WalletWasabi/WalletWasabi/assets/16364053/3e2d8c56-3518-4bb3-ae35-6ff790424687)

ichthus1604 commented:
@soosr 

> Select all quick action is missing.

Done ;)

> The Automatic icon is different 

Fixed ;)

soosr commented on WalletWasabi.Fluent/Controls/ContentArea.axaml:
```diff
@@ -27,18 +27,18 @@
           <DockPanel Name="MainDockPanel" IsVisible="{Binding !#LoadingRing.IsVisible}">
 
-            <Panel DockPanel.Dock="Bottom" x:Name="PART_ButtonArea">
+            <DockPanel DockPanel.Dock="Bottom" x:Name="PART_ButtonArea" Margin="31 0 31 10">
```
The issue with the margin is that you should also change the type in style, so please remove the margin set from the control, and update the style.

https://github.com/WalletWasabi/WalletWasabi/blob/37537da0eb9ed7cccd395a3d8884a2fc1431dee4/WalletWasabi.Fluent/Controls/ContentArea.axaml#L121

ichthus1604 replied:
Fixed

soosr commented on WalletWasabi.Fluent/Icons/Icons.axaml:
```diff
@@ -206,6 +206,14 @@
       <StreamGeometry x:Key="coordinator">M9.25 6C8.83579 6 8.5 6.33579 8.5 6.75C8.5 7.16421 8.83579 7.5 9.25 7.5H14.75C15.1642 7.5 15.5 7.16421 15.5 6.75C15.5 6.33579 15.1642 6 14.75 6H9.25ZM8.5 17.75C8.5 17.3358 8.83579 17 9.25 17H14.75C15.1642 17 15.5 17.3358 15.5 17.75C15.5 18.1642 15.1642 18.5 14.75 18.5H9.25C8.83579 18.5 8.5 18.1642 8.5 17.75ZM9.25 14C8.83579 14 8.5 14.3358 8.5 14.75C8.5 15 ...
+      <StreamGeometry x:Key="manual_control">F1 M3.88411 4.01327C3.74935 3.85155 3.52773 3.79156 3.32979 3.86322C3.13185 3.93489 3 4.12285 3 4.33336L3 8.66667C3 8.94287 3.22386 9.16667 3.5 9.16667C3.77614 9.16667 4 8.94287 4 8.66667L4 5.7144L5.61587 7.65347C5.71087 7.76747 5.8516 7.83333 6 7.83333C6.1484 7.83333 6.28913 7.76747 6.38413 7.65347L8 5.7144L8 8.66667C8 8.94287 8.22387 9.16667 8.5 9.16 ...
+
+      <DrawingImage x:Key="automatic_control">
+        <DrawingGroup>
+          <GeometryDrawing Brush="{DynamicResource TextForegroundColor}" Geometry="F1 M2.16667 0.5C0.973333 0.5 0 1.47333 0 2.66667L0 10.3333C0 11.5267 0.973333 12.5 2.16667 12.5L9.83333 12.5C11.0267 12.5 12 11.5267 12 10.3333L12 2.66667C12 1.47333 11.0267 0.5 9.83333 0.5L2.16667 0.5ZM1 2.66667C1 2.02 1.52 1.5 2.16667 1.5L9.83333 1.5C10.48 1.5 11 2.02 11 2.66667L11 10.3333C11 10.98 10.48 11.5 9.8 ...
+          <GeometryDrawing Brush="{DynamicResource TextForegroundColor}" Geometry="F1 M8.78696 8.44673L6.45362 3.61339C6.28696 3.26673 5.72029 3.26673 5.55362 3.61339L3.22029 8.44673C3.10029 8.69339 3.20696 8.99339 3.45362 9.11339C3.70696 9.23339 4.00029 9.12673 4.12029 8.88006L4.54696 7.99339L7.46029 7.99339L7.88696 8.88006C7.97362 9.06006 8.15362 9.16006 8.34029 9.16006C8.41362 9.16006 8.48696  ...
+        </DrawingGroup>
+      </DrawingImage>
```
If you export the icon as SVG from figma, you can get the path from it, use only a `StreamGeometry` please.

```
M3.25 0C1.46 0 0 1.46 0 3.25V14.75C0 16.54 1.46 18 3.25 18H14.75C16.54 18 18 16.54 18 14.75V3.25C18 1.46 16.54 0 14.75 0H3.25ZM1.5 3.25C1.5 2.28 2.28 1.5 3.25 1.5H14.75C15.72 1.5 16.5 2.28 16.5 3.25V14.75C16.5 15.72 15.72 16.5 14.75 16.5H3.25C2.28 16.5 1.5 15.72 1.5 14.75V3.25Z M13.1799 11.92L9.67995 4.67003C9.42995 4.15003 8.57995 4.15003 8.32995 4.67003L4.82995 11.92C4.64995 12.29 4.80995 12.74 5.17995 12.92C5.55995 13.1 5.99995 12.94 6.17995 12.57L6.81995 11.24H11.1899L11.8299 12.57C11.9599 12.84 12.2299 12.99 12.5099 12.99C12.6199 12.99 12.7299 12.97 12.8399 12.92C13.2099 12.74 13.3699 12.29 13.1899 11.92H13.1799ZM7.53995 9.75003L8.99995 6.73003L10.4599 9.75003H7.53995Z
```

ichthus1604 replied:
Fixed.

soosr commented on WalletWasabi.Fluent/ViewModels/Wallets/Send/SendViewModel.cs:
```diff
@@ -58,26 +60,35 @@ public partial class SendViewModel : RoutableViewModel
+			: _walletModel.Balances;
 
-		_suggestionLabels = new SuggestionLabelsViewModel(WalletVm.WalletModel, Intent.Send, 3);
+		BalanceCaption =
+			_parameters.IsManual
+			? "Max:"
+			: "Balance:";
```
It can be always `Max` regardless of the mode.

ichthus1604 replied:
Fixed.

soosr commented on WalletWasabi.Fluent/WalletWasabi.Fluent.csproj:
```diff
@@ -38,4 +38,10 @@
 		<RuntimeHostConfigurationOption Include="System.Globalization.Invariant" Value="true" />
 	</ItemGroup>
+
+	<ItemGroup>
+	  <Compile Update="Views\Wallets\Send\ManualControlDialogView.axaml.cs">
+	    <DependentUpon>ManualControlDialogView.axaml</DependentUpon>
+	  </Compile>
+	</ItemGroup>
```
revert

ichthus1604 replied:
Fixed.

soosr reviewed (changes requested):
- In the scenario when exchange rate is not available, the USD amount shouldn't should be visible.
![image](https://github.com/WalletWasabi/WalletWasabi/assets/16364053/d73d3e05-ac71-4518-813e-bf356dd9e589)

soosr reviewed (changes requested):
Not yet fixed: https://github.com/WalletWasabi/WalletWasabi/pull/12888#pullrequestreview-2114790552

ichthus1604 commented:
@soosr 

> In the scenario when exchange rate is not available, the USD amount shouldn't should be visible.

Fixed.

turbolay reviewed (approved):
Tested various scenarios, everything seems to work fine.
Feature is really good.

I can see 2 slight UXs improvement to make in future PRs:
- On `MainViewModel`, remember last button chosen as a default (I can see that users will usually want to follow the same workflow)
- Auto-fill the amount to max amount. This simply makes sense because it incentives to send the full coins

I would also suggest to later remove `CTRL+C+D/Wallet Coins`. This feature is straight useless now as you can have the exact same information here. We should also remove `Coin Control`,
